solver/llbsolver/file/backend.go

```diff
@@ -205,15 +205,31 @@ func docopy(ctx context.Context, src, dest string, action pb.FileActionCopy, u *
 		copy.WithXAttrErrorHandler(xattrErrorHandler),
 	}
 
-	if !action.AllowWildcard {
+	copy_one := func(s string) error {
```

Comment:
Would rather not have an inline function here, I think we could do this more neatly by reversing the condition and having an `if action.AllowWildcard` later on, and doing the wildcard resolution there, appending to a list/map/etc.

Then we can simply end up with an object that we can iterate through at the end that does the actual copy.

Reply:
Coming from C++/C++ template metaprogramming world, idiomatic go and such is... hmm, strange :grin:
In C++ terms inline function approach might be preferred - inlining lambda is easier than optimizing out redundant push/pop compilation-wise... so you are just used to such a pattern.
Lyrics aside, following your lead - done.